app/controllers/ResourceIndex.java

```diff
@@ -105,17 +95,20 @@ public Result list(String q, int from, int size, String sort, boolean list)
 
     if (request().accepts("text/html")) {
       return ok(render("OER World Map", "ResourceIndex/index.mustache", scope));
-    } else if (request().accepts("text/csv")) {
+    } //
+    else if (request().accepts("text/csv")) {
       StringBuffer result = new StringBuffer();
```

Comment:
Yes, exactly! Building the entire export string should be done in the actual exporter, not the controller which should only call

`return ok(new CsvWithNestedIdsExporter().export(resourceList)).as("text/csv");`